### DISCUSSION TITLE
[@types/telebot]

### DISCUSSION THREAD

**Discussion by tennox** [DefinitelyTyped/DefinitelyTyped#56596] [Issues with a @types package]
- [x] I tried using the `@types/telebot` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).

I found out that methods are not typed as returning a promise (i.e. not marked as async) e.g. the `sendMessage` method. ([see in examples](https://github.com/mullwar/telebot/blob/a614fce30092b998fadd65af507fbc924c698fef/examples/edit-text.js#L6) how they return promises) 

I think all methods actually return Promises

[a comment by an automated account is not shown]

**Comment by tennox** [comment]
I have created a PR that I think addresses the relevant methods:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56597

**Comment by mariotsi** [comment]
Hello @tennox,
when I wrote the original implementation Telebot was callback based. Looking at your PR now
